backends/tfhe-cuda-backend/cuda/src/device.cu

```diff
@@ -52,184 +50,132 @@ void *cuda_malloc_async(uint64_t size, cuda_stream_t *stream) {
-    // error code: invalid device pointer
-    return -1;
+    printf("Cuda error: invalid device pointer in async copy to GPU.");
+    abort();
   }
 
-  cudaSetDevice(stream->gpu_index);
+  check_cuda_error(cudaSetDevice(stream->gpu_index));
   check_cuda_error(
       cudaMemcpyAsync(dest, src, size, cudaMemcpyHostToDevice, stream->stream));
-  return 0;
 }
 
-/// Tries to copy memory to the GPU synchronously
-/// 0: success
-/// -1: error, invalid device pointer
-/// -2: error, gpu index doesn't exist
-/// -3: error, zero copy size
-int cuda_memcpy_to_gpu(void *dest, void *src, uint64_t size) {
+/// Copy memory to the GPU synchronously
+void cuda_memcpy_to_gpu(void *dest, void *src, uint64_t size) {
   if (size == 0) {
-    // error code: zero copy size
-    return -3;
+    printf("Cuda error: zero copy size in copy to GPU.");
+    abort();
   }
-
   cudaPointerAttributes attr;
-  cudaPointerGetAttributes(&attr, dest);
+  check_cuda_error(cudaPointerGetAttributes(&attr, dest));
   if (attr.type != cudaMemoryTypeDevice) {
-    // error code: invalid device pointer
-    return -1;
+    printf("Cuda error: invalid device pointer in copy to GPU.");
+    abort();
   }
-
   check_cuda_error(cudaMemcpy(dest, src, size, cudaMemcpyHostToDevice));
-  return 0;
 }
 
-/// Tries to copy memory to the CPU synchronously
-/// 0: success
-/// -1: error, invalid device pointer
-/// -2: error, gpu index doesn't exist
-/// -3: error, zero copy size
-int cuda_memcpy_to_cpu(void *dest, void *src, uint64_t size) {
+/// Copy memory to the CPU synchronously
+void cuda_memcpy_to_cpu(void *dest, void *src, uint64_t size) {
   if (size == 0) {
-    // error code: zero copy size
-    return -3;
+    printf("Cuda error: zero copy size in cpoy to CPU.");
+    abort();
   }
-
   cudaPointerAttributes attr;
-  cudaPointerGetAttributes(&attr, src);
+  check_cuda_error(cudaPointerGetAttributes(&attr, src));
   if (attr.type != cudaMemoryTypeDevice) {
-    // error code: invalid device pointer
-    return -1;
+    printf("Cuda error: invalid device pointer in copy to CPU.");
+    abort();
   }
-
   check_cuda_error(cudaMemcpy(dest, src, size, cudaMemcpyDeviceToHost));
-  return 0;
 }
 
-/// Tries to copy memory within a GPU asynchronously
-/// 0: success
-/// -1: error, invalid device pointer
-/// -2: error, gpu index doesn't exist
-/// -3: error, zero copy size
-int cuda_memcpy_async_gpu_to_gpu(void *dest, void *src, uint64_t size,
+/// Copy memory within a GPU asynchronously
+void cuda_memcpy_async_gpu_to_gpu(void *dest, void *src, uint64_t size,
                                  cuda_stream_t *stream) {
   if (size == 0) {
-    // error code: zero copy size
-    return -3;
+    printf("Cuda error: zero copy size in copy from GPU to GPU.");
+    abort();
   }
 
-  if (stream->gpu_index >= cuda_get_number_of_gpus()) {
-    // error code: invalid gpu_index
-    return -2;
-  }
   cudaPointerAttributes attr_dest;
```

Comment:
gpu_index check gone ?

Reply:
This error is caught by check_cuda_error